<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Stefanos Natsis</author>
 <comment>Query builder for point cloud layers</comment>
 <class>QgsPointCloudQueryBuilderBase</class>
 <widget class="QDialog" name="QgsPointCloudQueryBuilderBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>727</width>
    <height>707</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Query Builder</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>.</normaloff>.</iconset>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QLabel" name="lblDataUri">
     <property name="text">
      <string>Datasource</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QSplitter" name="splitter_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QGroupBox" name="groupBox1">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="title">
        <string>Fields</string>
       </property>
       <layout class="QGridLayout">
        <property name="leftMargin">
         <number>11</number>
        </property>
        <property name="topMargin">
         <number>11</number>
        </property>
        <property name="rightMargin">
         <number>11</number>
        </property>
        <property name="bottomMargin">
         <number>11</number>
        </property>
        <item row="0" column="0">
         <widget class="QListView" name="lstAttributes">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;List of attributes in this point cloud&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="groupBox2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="title">
        <string>Values</string>
       </property>
       <layout class="QGridLayout">
        <property name="leftMargin">
         <number>11</number>
        </property>
        <property name="topMargin">
         <number>11</number>
        </property>
        <property name="rightMargin">
         <number>11</number>
        </property>
        <property name="bottomMargin">
         <number>11</number>
        </property>
        <item row="0" column="0" colspan="2">
         <widget class="QListView" name="lstValues">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;List of values for the current attribute.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="autoFillBackground">
           <bool>true</bool>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="uniformItemSizes">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="verticalLayoutWidget">
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QgsCollapsibleGroupBox" name="groupBox4">
         <property name="title">
          <string>Operators</string>
         </property>
         <layout class="QGridLayout">
          <property name="leftMargin">
           <number>11</number>
          </property>
          <property name="topMargin">
           <number>11</number>
          </property>
          <property name="rightMargin">
           <number>11</number>
          </property>
          <property name="bottomMargin">
           <number>11</number>
          </property>
          <item row="0" column="6">
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="btnLessThan">
            <property name="text">
             <string>&lt;</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="btnGreaterThan">
            <property name="text">
             <string>&gt;</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QPushButton" name="btnEqual">
            <property name="text">
             <string>=</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="btnLessEqual">
            <property name="text">
             <string>&lt;=</string>
            </property>
           </widget>
          </item>
          <item row="0" column="4">
           <widget class="QPushButton" name="btnIn">
            <property name="text">
             <string>IN</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="btnGreaterEqual">
            <property name="text">
             <string>&gt;=</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="btnNotEqual">
            <property name="text">
             <string>!=</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QPushButton" name="btnAnd">
            <property name="text">
             <string>AND</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QPushButton" name="btnOr">
            <property name="text">
             <string>OR</string>
            </property>
           </widget>
          </item>
          <item row="1" column="4">
           <widget class="QPushButton" name="btnNotIn">
            <property name="text">
             <string>NOT IN</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox3">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Provider Specific Filter Expression</string>
         </property>
         <layout class="QGridLayout">
          <property name="leftMargin">
           <number>11</number>
          </property>
          <property name="topMargin">
           <number>11</number>
          </property>
          <property name="rightMargin">
           <number>11</number>
          </property>
          <property name="bottomMargin">
           <number>11</number>
          </property>
          <item row="0" column="0">
           <widget class="QgsCodeEditorSQL" name="mTxtSql" native="true"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
  <zorder>splitter_2</zorder>
  <zorder>lblDataUri</zorder>
  <zorder>buttonBox</zorder>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QgsCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>qgscollapsiblegroupbox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsCodeEditorSQL</class>
   <extends>QWidget</extends>
   <header>qgscodeeditorsql.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>lstAttributes</tabstop>
  <tabstop>btnEqual</tabstop>
  <tabstop>btnLessThan</tabstop>
  <tabstop>btnGreaterThan</tabstop>
  <tabstop>btnIn</tabstop>
  <tabstop>btnLessEqual</tabstop>
  <tabstop>btnGreaterEqual</tabstop>
  <tabstop>btnNotEqual</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>QgsPointCloudQueryBuilderBase</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>530</x>
     <y>549</y>
    </hint>
    <hint type="destinationlabel">
     <x>674</x>
     <y>480</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>QgsPointCloudQueryBuilderBase</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>641</x>
     <y>553</y>
    </hint>
    <hint type="destinationlabel">
     <x>672</x>
     <y>521</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
